pkgs/development/python-modules/pyinfra/default.nix

```
@@ -6,29 +6,33 @@
   configparser,
   distro,
   fetchFromGitHub,
+  fetchpatch,
   gevent,
   jinja2,
+  packaging,
   paramiko,
   pytestCheckHook,
   python-dateutil,
   pythonOlder,
   pywinrm,
   pyyaml,
   setuptools,
+  typeguard,
+  typing-extensions,
 }:
 
 buildPythonPackage rec {
   pname = "pyinfra";
-  version = "2.9.2";
+  version = "3.0";
   format = "setuptools";
```

```suggestion
  pyproject = true;
```

Migrate from the legacy setuptools format to a PEP517 build.